### PR TITLE
bug: make weave_to_flannel idempotent

### PR DESCRIPTION
**Pull request by ricardomaraschini**
#### What this PR does / why we need it:

This PR contains some improvements on the `weave_to_flannel` migration function. We want to guarantee that if something goes wrong with the migration we will try it once more when the script runs again and that it will eventually finish its work correctly.

#### Notes

Depending where the Weave to Flannel migration fails we may end up with a cluster in a unhealthy state (no CNI running thus some pods may be failing). As the cluster may not be in a healthy state after the failure the user may need to provide the `-s host-preflight-ignore` flag to the script during a second attempt.

The second attempt to migrate from weave to flannel is dispatched directly from the flannel pre_init phase, this is done because we can't guarantee that all addons preflights will succeed in case of a cluster running without CNI. 

#### About testing

In order to test this code I had to randomly `ctrl+c` the installer during different stages of the weave to flannel migration and then re-run the install script passin gin `-s host-preflight-ignore` flag. I have tested this in both single and multi nodes scenarios.

### CODE REVIEW

File: addons/flannel/0.21.5/install.sh [hunk 1]
```diff
@@ -295,6 +317,10 @@ function flannel_scale_up_rook() {
 function weave_to_flannel() {
     local dst="$DIR/kustomize/flannel"
 
+    # start by creating a config map indicating that this migration has been started. if we fail
+    # midair this ensure this function will be ran once more.
+    kubectl create configmap -n kube-system migrating-from-weave --from-literal=started="true" --dry-run=client -o yaml | kubectl apply -f -
```
Comment (camilamacedo86) [inline addons/flannel/0.21.5/install.sh:322]: Currently, we have already `kurl-migration-from-rook` see https://github.com/replicatedhq/kURL/blob/a99174144c2aa8a26717b48709ea975cddf44456/scripts/common/rook.sh#L419

Could we follow the same standard and keep the configmap in the kurl namespace since it is a kurl thing?

File: scripts/common/preflights.sh [hunk 2]
```diff
@@ -30,10 +30,15 @@ function init_preflights() {
 # if kurl pods like ekco not be running then we should bail
 function bail_if_kurl_pods_are_unhealthy() {
     if commandExists kubectl; then
-      log "Awaiting 2 minutes to check kURL Pod(s) are Running"
-      if ! spinner_until 120 check_for_running_pods kurl; then
-          bail "Kurl has unhealthy Pod(s). Check the namespace kurl. Restarting the pod may fix the issue."
-      fi
+        # if host preflight are being ignored we just move forward.
```
Comment (camilamacedo86) [inline scripts/common/preflights.sh:33]: Should not it be in another PR so that we can either generate the release notes accordingly to know what is fixed or not in each version which makes easier when we need to gathering this info in the support

> Fixes pods check to allow ignore it failures when HOST_PREFLIGHT_IGNORE flag is used

Comment (ricardomaraschini) [reply scripts/common/preflights.sh:33]: For sake of simplicity I am keeping this here.

File: addons/flannel/0.21.5/install.sh (hunk 1 above)
Comment (ricardomaraschini) [reply addons/flannel/0.21.5/install.sh:335]: Done.

**Review by rrpolanco — APPROVED** [review]
LGTM. We should also add a story to fix the test **weave to flannel, docker to containerd, single node** on Rocky Linux.

**Review by camilamacedo86 — APPROVED** [review]
